manifests/mail_on_failure.pp

```diff
@@ -0,0 +1,43 @@
+# mail on systemd units failing
+#
+# This class manages a set of scripts that helps to
+# send out emails of failing units.
+# It prepares a unit template that can then be referenced
+# in units via OnFailure=status_email_USER@%n.service
+#
+# Email sending is handed over to sendmail and expected to
+# be allowed for user nobody.
+#
+# This is helpful to monitor failing systemd.timers in a cron-like
+# MAILTO style reporting.
+# See https://wiki.archlinux.org/index.php/Systemd/Timers#MAILTO
+# for background and implementation details.
+#
+# @param email_users
+#   Users to prepare a template unit for.
+#   By default this class prepares a template for root.
+#   Besides local users you can also define an email address.
+#   Due to the naming restrictionsof systemd units, the `@` in the template name
+#   will be replaced with `_AT_` and `ALIAS@DOMAIN` will get a unit called:
+#   `status_email_ALIAS_AT_DOMAIN@%n.service`
+#
+# @api public
+#
+class systemd::mail_on_failure (
+  Array[String[1]] $email_users = ['root'],
+) {
+  file {
```

Comment:
Why the newline here?

Reply:
no particular reason. I often do it on a visually basis. Is there an official style guide when to do it and when not? Even in this module it seems to be mixed already.

Reply:
Afaik, I've only seen this style when there's multiple resources of the same type, separated by semi-colons.

Reply:
yeah, then you definitely wanna do it.

I don't have hard feelings about it. Just curious if there are official "rules"?

Reply:
There is an official style guide, yes:

"Must use one space between the resource type and opening brace, one space between the opening brace and the title, and no spaces between the title and colon."

https://puppet.com/docs/puppet/7.1/style_guide.html#spacing-indentation-and-whitespace

the only exception being [multiple resources in a declaration](https://puppet.com/docs/puppet/7.1/style_guide.html#resource-arrangement):

"Use semicolon-separated multiple resource bodies only in conjunction with a local default body."

Reply:
Awesome, thanks for looking that up.